File: Bisq/src/main/java/com/joachimneumann/bisq/Database/RawBisqNotificationDao.java
package com.joachimneumann.bisq.Database;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface RawBisqNotificationDao {
    @Query("SELECT * FROM rawBisqNotification")
    LiveData<List<RawBisqNotification>> getAll();

    @Insert
    void insert(RawBisqNotification rawBisqNotification);

    @Insert
    void insertAll(RawBisqNotification... rawBisqNotification);

    @Delete
    void delete(RawBisqNotification rawBisqNotification);
}

File: Bisq/src/main/java/com/joachimneumann/bisq/Database/RawBisqNotificationDao.kt
package com.joachimneumann.bisq.Database

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query

@Dao
interface RawBisqNotificationDao {
    @get:Query("SELECT * FROM rawBisqNotification")
    val all: LiveData<List<RawBisqNotification>>

    @Insert
    fun insert(rawBisqNotification: RawBisqNotification)

    @Insert
    fun insertAll(vararg rawBisqNotification: RawBisqNotification)

    @Delete
    fun delete(rawBisqNotification: RawBisqNotification)

    @Query("DELETE FROM rawBisqNotification")
    fun nukeTableRawBisqNotification()
}
